src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -165,8 +168,16 @@ module Make_update (T : Transaction_snark.Verification.S) = struct
         in
         result
       in
+      let%bind is_first_block =
+        previous_state |> Protocol_state.consensus_state
+        |> Consensus.Data.Consensus_state.is_genesis_state_var
+      in
+      let%bind genesis_state_hash =
```

Comment:
I think it is actually not necessary to special case the logic for the `is_first_block `case here. That is, I think we can always just say `new_state.genesis_state_hash == previous_state.genesis_state_hash`.

Reply:
But `new_state.genesis_state_hash == previous_state.genesis_state_hash` is not true on the first block since the compile time block we prove has a constant compile time `genesis_state_hash` that we don't want to keep, no?

Reply:
Oh right, this was when `negative_one` had a dummy ledger before I decided to have the proof as well generated at runtime. Thanks for catching that. I'll remove the condition

Reply:
Oops, this is still needed for the genesis proof because the genesis state hash of the `negative_one` is zero